sdk/storage/Azure.Storage.DataMovement/src/JobPlan/JobPlanFile.cs

```diff
@@ -58,9 +58,20 @@ public static async Task<JobPlanFile> CreateJobPlanFileAsync(
             string filePath = Path.Combine(checkpointerPath, fileName);
 
             JobPlanFile jobPlanFile = new(id, filePath);
-            using (FileStream fileStream = File.Create(jobPlanFile.FilePath))
+            try
```

Comment:
This is a little outside the scope of the PR but I feel like if there are issues with creating the JobPlanFile (and JobPartPlanFile), we should just delete the whole file cus it will be in a corrupted state